ember-server/shared/src/main/scala/org/http4s/ember/server/internal/WebSocketHelpers.scala

```diff
@@ -121,32 +122,28 @@ object WebSocketHelpers {
       val (stream, onClose) = ctx.webSocket match {
         case WebSocketCombinedPipe(receiveSend, onClose) =>
           incoming
-            .through(decodeFrames(frameTranscoder))
-            .through(handleIncomingFrames(write, frameTranscoder, close))
+            .through(decodeFrames[F])
+            .evalMapFilter(handleIncomingFrame[F](writeFrame, close))
             .through(receiveSend)
-            .through(encodeFrames(frameTranscoder))
-            .through(write) -> onClose
+            .foreach(writeFrame) -> onClose
         case WebSocketSeparatePipe(send, receive, onClose) =>
-          val closeFrame: F[Stream[F, WebSocketFrame]] = close.get.flatMap {
+          val closeFrame: F[List[WebSocketFrame]] = close.get.flatMap {
```

Comment:
Can we use an `Option` here?

Reply:
I am afraid not, because the `evalSeq` below needs an `Iterable`, and an Option is not.

Reply:
What if we just do `eval(...).unNone` or similar? `evalSeq` doesn't seem to be doing anything special.

```scala
  def evalSeq[F[_], S[A] <: Seq[A], O](fo: F[S[O]]): Stream[F, O] =
    eval(fo).flatMap(Stream.emits)
```

https://github.com/typelevel/fs2/blob/9e924ca24ef72d15b09825905ea29b344e4bd67b/core/shared/src/main/scala/fs2/Stream.scala#L3165-L3166

Reply:
Just pushed a change to wrap the `sendFrame` into the method instead, and leave a `Strea.eval` for it.